### DISCUSSION TITLE
regarding the future of this wrapper

### DISCUSSION THREAD

**Discussion by sombriks** [sombriks/node-libgpiod#8] [Ideas]
Hello there!

- General Kenno-- oh wait sorry 😅 

Well, @dmitrydvorkin raise the ball about simply ditch libgpiod for good and speak directly with kernel calls.

Now we're a group of three and i think we could debate this.

Of course no rush, once it's opensource and we're kinda just knowing each other at this point.

My initial idea on this project was to be the most simple, straightforward and easy wrapper for what kernel people says is the right  way to interface with gpio chips.

@noctarius i would hear, at some moment, what do you think about it.

I am not laying any opinion right now, because we don't have communication with libgpiod maintainer right now. And it might be important.

so far this is what we can say right now:

- the lib has the right name. ppl looking for modern gpio will hit here because kernel people says so

Please at some convenient moment let's speak lightly about it, about the sex of the angels, and other blinky issues. No Rush.

It might be a question not only about codesign excellence, but about 'marketing'. Weird, i know, but this is my opinion and your points of view are important equally!

Sorry my language at some point, i am a bit inebriated by [cachaça](https://en.wikipedia.org/wiki/Ypi%C3%B3ca_Group)

Heyy you guys are awesome!!! Let's put nodejs everywhere hahahaha"!

**Comment by noctarius** [comment]
As mentioned in the other ideas thread, I don't really care if it stays as a wrapper or if we call kernel apis directly, except that the later requires us to stay up to date with changes in those ioctls, whereas libgpiod takes care of that.

**Reply by dmitrydvorkin** [reply]
in case of using direct "kernel" API you don't have to care about structures alloc/free ops. Everything is done on FD-base.
It will make the API more stable and get rid of SEGVs.
I suspect, the compatibility will exist for the long time, also there are traditional #ifdef way out in C/C++ for that.
